subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -1069,6 +1187,30 @@ static bool send_request(enum http_method method,
 		}
 
 		break;
+
+	case HAWKBIT_DOWNLOAD_PART:
```

Comment:
The difference of `HAWKBIT_DOWNLOAD_PART` from the usual `HAWKBIT_DOWNLOAD` is basically the addition of `Range` header and a 'tweaked' version of the response's `HAWKBIT_DOWNLOAD` handling to set the `final_data_received` flag when it has downloaded all of the firmware.

Do you think if it is possible to streamline this a little such that there's only one type of download, with the `Range` header there as default, Kconfig is used to limit the download size, if defined, and do a similar loop that you currently have. Then the `response_cb` will handle the response differently, especially the `HTTP_DATA_FINAL`, depending on the Kconfig.

Then we will continue to have a single response_cb to handle the download datasteam, lesser code duplication, and easier to implement resumable download.

Reply:
@Navin-Sankar any thoughts?